system/jlib/jthread.cpp

```diff
@@ -1840,6 +1842,8 @@ protected: friend class PipeWriterThread;
         if (et)
         {
             et->stop();
+            if (clearStderr)
+                delete et;
             // NOTE - we don't delete it here, since we want to be able to still read the buffered data
```

Comment:
slightly pedantic, but this comment should be moved down to the call from wait() - it is a bit confusing here.